### ISSUE TITLE
Check if model is fine-tunable

### ISSUE THREAD

**Issue by wochinge** [RasaHQ/rasa#7330]
Before we can [finetune a model](https://github.com/RasaHQ/rasa/issues/6971), we need to make sure that this is actually possible.

This is the list of constraints defined [here](https://docs.google.com/document/d/1B7iVPRQONPoFssJfxhLMm5o23KWTPYNPQw_7XRc38Ao/edit):
- Model config of the previous checkpoint should be exactly the same as the new config, except for the number of epochs of ML components like `DIETClassifier`, `ResponseSelector` and `TEDPolicy`
- The labels for each ML component should also be the same between the two versions of data the models are trained on 
  - labels: entity names, intent names, action names
  - they also need the same order!
- Previous checkpoint trained with the “MINIMUM_COMPATIBLE_VERSION” of Rasa Open Source.

**Comment by joejuzl** [comment]
> The labels for each ML component should also be the same between the two versions of data the models are trained on
labels:
>  - entity names, intent names, action names 
>  - they also need the same order!

Is there anything in `domain.yml` that we _are_ able to change?
